wranglertools/import_data.py

```diff
@@ -154,65 +153,76 @@ def md5(path):
-            except requests.exceptions.MissingSchema:
-                print("\nERROR : The 'attachment' field contains INVALID FILE PATH or URL ({})\n".format(path))
-                sys.exit(1)
-            # if it works as a URL, but does not return 200
-            if r.status_code is not 200:  # pragma: no cover
-                print("\nERROR : The 'attachment' field contains INVALID URL ({})\n".format(path))
-                sys.exit(1)
+            except Exception:
+                raise Exception("\nERROR : The 'attachment' field has INVALID FILE PATH or URL ({})\n".format(path))
+            else:
+                # if it works as a URL, but does not return 200
+                if r.status_code is not 200:  # pragma: no cover
+                    raise Exception("\nERROR : The 'attachment' field has INVALID URL ({})\n".format(path))
             # parse response
             path = path.split("/")[-1]
-            with open(path, "wb") as outfile:
-                outfile.write(r.content)
-
-    filename = os.path.basename(path)
-    mime_type = mimetypes.guess_type(path)[0]
-    major, minor = mime_type.split('/')
-    detected_type = magic.from_file(path, mime=True)
-    # XXX This validation logic should move server-side.
-    if not (detected_type == mime_type or
-            detected_type == 'text/plain' and major == 'text'):
-        if not (minor == 'zip' or major == 'text'):  # zip files are special beasts
-            raise ValueError('Wrong extension for %s: %s' % (detected_type, filename))
+            try:
+                with open(path, "wb") as outfile:
+                    outfile.write(r.content)
+                    ftp_attach = True
+            except Exception as e:
+                raise Exception("\nERROR : Cannot write a tmp file to disk - {}".format(e))
+
+    allowed_mimes = (
+        'application/pdf',
+        'application/zip',
+        'text/plain',
+        'text/tab-separated-values',
+        'text/html',
+        'application/msword',
+        'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
+        'application/vnd.ms-excel',
+        'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+        'image/png',
+        'image/jpeg',
+        'image/gif',
+        'image/tiff',
+    )
     attach = {}
+    filename = os.path.basename(path)
+    guessed_mime = mimetypes.guess_type(path)[0]
+    detected_mime = magic.from_file(path, mime=True)
+    # NOTE: this whole guesssing and detecting bit falls apart for zip files which seems a bit dodgy
+    # some .zip files are detected as generic application/octet-stream but don't see a good way to verify
+    # basically relying on extension with a little verification by magic for most file types
+    if detected_mime != guessed_mime and guessed_mime != 'application/zip':
+        raise ValueError('Wrong extension for %s: %s' % (detected_mime, filename))
+    if guessed_mime not in allowed_mimes:
```

Comment:
I would check this before the above - the above is not worth checking if the below is not true, so check it first.